extensions/src/test/java/org/pentaho/platform/plugin/services/email/EmailServiceTest.java

```diff
@@ -0,0 +1,285 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ *
+ * Copyright (c) 2002-2023 Hitachi Vantara. All rights reserved.
+ *
+ */
+
+package org.pentaho.platform.plugin.services.email;
+
+import junit.framework.TestCase;
+import org.apache.commons.lang.ObjectUtils;
+import org.apache.http.HttpException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.pentaho.di.core.encryption.Encr;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Properties;
+
+@RunWith( MockitoJUnitRunner.class )
+public class EmailServiceTest extends TestCase {
+
+  private EmailService emailService = null;
+
+  private File defaultConfigFile = null;
+  @Before
+  public void setUp() throws Exception {
+    defaultConfigFile = File.createTempFile( "email_config_", ".xml" );
+    this.emailService = new EmailService( defaultConfigFile );
+  }
+
+
+  @Test( expected = HttpException.class )
+  public void testGetOAuthToken() throws HttpException, IOException {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "tes@test.com", "Pentaho Scheduler", "", 25,
+                    "graph_api", true, "test@test.com", "", true, true );
+    emailConfigOriginal.setAuthMechanism( "XOAUTH2" );
+    emailConfigOriginal.setClientId( "cid" );
+    emailConfigOriginal.setClientSecret( "secret" );
+    emailConfigOriginal.setTokenUrl( "https://login.microsoftonline.com/common/oauth2/v2.0/token" );
+    emailConfigOriginal.setScope( "https://graph.microsoft.com/.default" );
+    emailConfigOriginal.setGrantType( "client_credentials" );
+    emailService.getOAuthToken( emailConfigOriginal );
+  }
+
+  @Test( expected = HttpException.class )
+  public void testSendMailGraphApi() throws HttpException, IOException {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "tes@test.com", "Pentaho Scheduler", "", 25,
+                    "graph_api", true, "test@test.com", "", true, true );
+    emailConfigOriginal.setAuthMechanism( "XOAUTH2" );
+    emailConfigOriginal.setClientId( "cid" );
+    emailConfigOriginal.setClientSecret( "secret" );
+    emailConfigOriginal.setTokenUrl( "https://login.microsoftonline.com/common/oauth2/v2.0/token" );
+    emailConfigOriginal.setScope( "https://graph.microsoft.com/.default" );
+    emailConfigOriginal.setGrantType( "client_credentials" );
+    emailService.sendMailGraphApi( emailConfigOriginal, "token", "message" );
+  }
+
+  @Test( expected = HttpException.class )
```

Comment:
I'm not sure what the difference in the test is between `testSendMailGraphApi()` and `testSendMailGraphApiAuthCode()` because the code for EmailService doesn't have any logic dealing with auth codes. And you aren't (or shouldn't) be communicating with a real msgraph gateway because it's a unit test and there isn't any setup code.